app/models/piece.rb

```diff
@@ -69,6 +78,35 @@ def pawn_has_moved?
     end
   end
 
+  def castle?(rook_x_pos, rook_y_pos)
```

Comment:
A method with the same name and parameters has already been defined in this script

Reply:
what had happened was. I needed the method to exist in the piece model so that the rook and the king could both call it, i forgot to remove it from the king model but it is done now.